src/App.tsx

```diff
@@ -1,24 +1,168 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { addTodo, getTodos, removeTodo } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Todolist } from './components/Todolist';
+import { filterValues } from './constants';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6438;
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [errorType, setErrorType] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [isTodoAdding, setIsTodoAdding] = useState(false);
+  const [isTodoRemoving, setIsTodoRemoving] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState('All');
+  const [removingTodoIds, setRemovingTodoIds] = useState<number[]>([]);
+
+  const hasActive = todos.some(todoItem => !todoItem.completed);
+
+  const getTodosFromServer = async () => {
+    try {
+      const response = await getTodos(USER_ID);
+
+      setTodos(response);
+    } catch (error) {
+      setHasError(true);
+      setErrorType('upload');
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = todos.filter(todo => {
+    if (selectedFilter === filterValues.completed) {
+      return todo.completed;
+    }
+
+    if (selectedFilter === filterValues.active) {
+      return !todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleAddTodo = async () => {
+    try {
+      setIsTodoAdding(true);
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      await addTodo(newTodo);
+
+      const demoTodo = {
+        ...newTodo,
+        id: 0,
+      };
+
+      setTempTodo(demoTodo);
+
```

Comment:
pay attention. now you set temp todo after a response from the server, but you should do it immediately after Enter was pressed

Reply:
Thank you for the review. Implemented all of that changes